service/s3.cc

```diff
@@ -2761,11 +2723,7 @@ std::shared_ptr<S3Api> getS3ApiForUri(const std::string & uri)
         accessKey = getAwsAccessKey(accessKeyId);
     }
 
-    auto api = make_shared<S3Api>(accessKeyId, accessKey);
```

Comment:
@jeremybarnes also.... to clarify the context of this change : we have the above problems with real customers, so it's a very practical issue here. I would be a bit reluctant to ask them to adjust their permissions to accommodate our s3 code.